src/Hazelcast.Net/HazelcastOptions.Build.cs

```diff
@@ -56,29 +58,32 @@ internal static HazelcastOptions Build(Action<IConfigurationBuilder> setup, Acti
         /// override them. This allows one json file to contain several configuration sets, which is
         /// convenient for instance when using the "user secrets" during tests.</para>
         /// </remarks>
-        internal static HazelcastOptions Build(Action<IConfigurationBuilder> setup, Action<IConfiguration, HazelcastOptions> configure, string altKey)
+        internal static HazelcastOptions Build(Action<IConfigurationBuilder> setup, Action<HazelcastOptions> preConfigure, Action<IConfiguration, HazelcastOptions> configure, string altKey)
         {
             if (setup == null) throw new ArgumentNullException(nameof(setup));
 
             var builder = new ConfigurationBuilder();
             setup(builder);
             var configuration = builder.Build();
 
-            return Build(configuration, configure, altKey);
+            return Build(configuration, preConfigure, configure, altKey);
         }
 
         // builds options, no alternate keys
-        private static HazelcastOptions Build(IConfiguration configuration, Action<IConfiguration, HazelcastOptions> configure = null)
-            => Build(configuration, configure, null);
+        private static HazelcastOptions Build(IConfiguration configuration, Action<HazelcastOptions> preConfigure, Action<IConfiguration, HazelcastOptions> configure = null)
+            => Build(configuration, preConfigure, configure, null);
 
         // builds options, optionally binding alternate keys
-        private static HazelcastOptions Build(IConfiguration configuration, Action<IConfiguration, HazelcastOptions> configure, string altKey)
+        private static HazelcastOptions Build(IConfiguration configuration, Action<HazelcastOptions> preConfigure, Action<IConfiguration, HazelcastOptions> configure, string altKey)
         {
             // must HzBind here and not simply Bind because we use our custom
             // binder which handles more situations such as ignoring and/or
             // renaming properties
 
             var options = new HazelcastOptions();
+
+            preConfigure?.Invoke(options);
```

Comment:
`The HazelcastOptionsBuilder and HazelcastOptions.Build have been extended to support pre-options. This allows the user to programmatically set options that still can be overriden by, say, command-line arguments or environment variables.`
Does this mean that you are changing the behaviour of user api options in a patch release? We usually tend not to change it even for minor releases.

Reply:
Somehow. I am extending the API so code that works in 5.0.0 cannot break due to 5.0.1 changes. OTOH new methods become available. Can we consider this OK? Else, will remove it 'til 5.1. Let me know.

Reply:
Yes, it can not be in 5.0.1. Can it be in 5.1? Let's discuss this. We normally do not do backward incompatible changes. The only way is to deprecate the old but keep the old.

Reply:
well, that is, I believe what I have described: I am adding a new method, but not removing any old method, nor changing the old behavior in any way. so 5.0.1 would be fully backward-compatible, wrt options, only it would provide an additional method.

Reply:
OK let me know, I am willing to remove that code, which is only used in an example, if that can help merging the PR.

Reply:
Code removed.